pymammotion/aliyun/cloud_gateway.py

```diff
@@ -427,12 +432,32 @@ def check_or_refresh_session(self):
         logger.debug(response.status_code)
         logger.debug(response.body)
 
-        # self._region = response.body.data
-        # Decodifica il corpo della risposta
+        # Decode the response body
         response_body_str = response.body.decode("utf-8")
 
-        # Carica la stringa JSON in un dizionario
-        json.loads(response_body_str)
+        # Load the JSON string into a dictionary
+        response_body_dict = json.loads(response_body_str)
+
+        if int(response_body_dict.get("code")) != 200:
+            raise Exception("Error check or refresh token: " + response_body_dict.get('msg', ''))
+
+        identityId = response_body_dict.get('data', {}).get('identityId', None)
```

Comment:
There should be a dataclass for this, so you can do e.g `IotToken().from_dict(response_body_dict.get('data', {}))`

Or create one, plenty of examples in the make-em-mow branch